shm/Alcohol.cpp

```diff
@@ -0,0 +1,56 @@
+#include "Alcohol.hpp"
+
+#include <iostream>
+#include <typeinfo>
+
+Alcohol::Alcohol(const std::string& name, size_t amount, size_t basePrice, Time* time)
+    : Alcohol(name, amount, basePrice, MAX_POWER, time) {
+}
+
+Alcohol::Alcohol(const std::string& name, size_t amount, size_t basePrice, size_t power, Time* time)
+    : Cargo(name, amount, basePrice, time),
+      _power(power) {
+}
+
+size_t Alcohol::getPrice() const {
+    return _basePrice * (_power / MAX_POWER);
```

Comment:
So all of those are size_t(unsigned integer). That means that if _power is lower than MAX_POWER it will round to zero making whole expression zero. You should do here some static casting to double to avoid this problem.

Reply:
+1